internal/coreinternal/scraperinttest/metrics.go

```diff
@@ -0,0 +1,29 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scraperinttest // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/scraperinttest"
+
+import (
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+)
+
+func EqualsLatestMetrics(expected pmetric.Metrics, sink *consumertest.MetricsSink, compareOpts []pmetrictest.CompareMetricsOption) func() bool {
+	return func() bool {
+		allMetrics := sink.AllMetrics()
+		return len(allMetrics) > 0 && nil == pmetrictest.CompareMetrics(expected, allMetrics[len(allMetrics)-1], compareOpts...)
```

Comment:
What if the sink is empty? I think this will panic

Reply:
It should be fine, I made similar changes here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21528/files

Reply:
I missed the `len(consumer.AllMetrics()) > 0` check 🤦‍♂️

Reply:
I don't think it's something that would vary per-run. It'd only be nil if the dev completely misused the function, in which case it'll panic every time until they fix it.

In any case, I think this function will get pulled into something that fully manages the sink - just trying to make incremental progress for now.